### PR TITLE
Assignment-5: React JavaScript

### PR DESCRIPTION
**Pull request by vaibhav-webonise**
**Problem statement:** Create todolist application

### CODE REVIEW

File: todolisApp/AddTodo.js [hunk 1]
```diff
@@ -0,0 +1,10 @@
+import React from "react";
+export const AddTodo = (props) => {
+  
+  return (
+    <div>
+      <label for="description">Enter todo </label><input type="text" name="description" value={props.desc} onChange={props.onUserType}/>
+      <input onClick={props.onAddTodo} type="button" id="addButton" value="Add todo" />
```
Comment (deepakb-webonise) [inline todolisApp/AddTodo.js]: hardcoded Add Todo, on update label should be changed.

File: todolisApp/ListTodo.js [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+            <th>Operation</th>
+          </tr>
+        </thead>
+        <tbody>
+          {props.todos.map(todoObject => (
+            <tr key={todoObject.todoId}>
+              <td>{todoObject.todoId}</td>
+              <td>{todoObject.description}</td>
+              <td>
+                <button onClick={props.onEdit.bind(null,todoObject.description,todoObject.todoId)}>Edit</button>
```
Comment (deepakb-webonise) [inline todolisApp/ListTodo.js]: no need to pass all todo data, only todo is enough.

File: todolisApp/Todo.js [hunk 3]
```diff
@@ -0,0 +1,84 @@
...
+
+    onAddTodo = () => {
+        if(this.state.description===''){
+            alert('Field can not be empty...');
+        }
+
+        else{
+            if(this.state.isUpdate){
+                let tempTodos = this.state.todosArray;
+                for(let index=0;index<tempTodos.length;index++) {
```
Comment (deepakb-webonise) [inline todolisApp/Todo.js]: use map over for loop

File: todolisApp/Todo.js [hunk 4]
```diff
@@ -0,0 +1,84 @@
...
+                    if(tempTodos[index].todoId==this.state.todoEditId) {
+                        tempTodos[index].description = this.state.description;
+                    }
+                }
+                this.setState({
+                    todosArray : tempTodos,
+                    description:"",
+                    isUpdate:false
+                })
+                return;
```
Comment (deepakb-webonise) [inline todolisApp/Todo.js]: no need of this statement

File: todolistApi/Todo.js [hunk 5]
```diff
@@ -0,0 +1,129 @@
...
+                axios({
+                    method: "put",
+                    url: `http://localhost:5000/todos/${this.state.editId}`,
+
+                    data: {
+                        desc: this.state.desc
+                    }
+                }).then((response) => {
+                    let tempTodos = this.state.todos;
+                    for (let index = 0; index < tempTodos.length; index++) {
```
Comment (deepakb-webonise) [inline todolistApi/Todo.js:54]: use map over for loop.

**Review by deepakb-webonise — APPROVED** [review]

**Review by swapnil-webonise — APPROVED** [review]

**Pull request merged**